File: T2/python/briot_ruffini.py
#!/usr/bin/env python3

def briot_ruffini(func, x):
    n = len(func)
    b = [None] * n
    for i in range(n):
        b[0] = func[0]
        for j in range(1, n):
            b[j] = x * b[j-1] + func[j]
        n -= 1
        func = b

    r = b[1]

    return r

File: T3/briot_ruffini.m
function y = briot_ruffini(n, a, x)
    Nx = length(x);
    for k = 1 : Nx
        b(1) = a(1);
        for i = 2 : n + 1
            b(i) = a(i) + x(k) * b(i-1);
        end
        y(k) = b(n+1);
    end
end

